docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,403 @@
+  .setTimeout(30)
+  .addOperation(
+    StellarSdk.Operation.invokeContractFunction({
+      function: "increment",
+    }),
+  )
+  .build();
+
+rpcServer.simulateTransaction(transaction).then((sim) => {
+  console.log("cost:", sim.cost);
+  console.log("result:", sim.result);
+  console.log("error:", sim.error);
+  console.log("latestLedger:", sim.latestLedger);
+});
+```
+
+## Understanding the Footprint
+
+A footprint is a set of ledger keys that the transaction will read or write. These keys are marked as either read-only or read-write:
+
+- **Read-Only Keys**: Available for reading only.
+- **Read-Write Keys**: Available for reading and writing.
+
+The footprint ensures that a transaction is aware of all the ledger entries it will interact with, preventing unexpected errors during execution.
+
+## Assembling a Transaction
+
+Once you have simulated the transaction and obtained the necessary data, you can assemble the transaction for actual submission. The `assembleTransaction` function in the SDK helps with this process, but you can also call `prepareTransaction` to have it both simulate and assemble the transaction for you.
+
+### js-stellar-sdk `assembleTransaction `
+
+Using the javascript SDK we can call [`assembleTransaction`](https://github.com/stellar/js-stellar-sdk/blob/master/src/rpc/transaction.ts#L28) to easily assemble a transaction.
+
+## Handling Archived Ledger Entries
+
+When a ledger entry is archived, it needs to be restored before the transaction can be submitted. This is indicated in the `restorePreamble` field of the result.
+
+```typescript
+interface RestorePreamble {
+  minResourceFee: string; // Recommended minimum resource fee to add when submitting the RestoreFootprint operation.
+  transactionData: string; // The recommended Soroban Transaction Data to use when submitting the RestoreFootprint operation.
+}
+```
+
+Here is an example for handling restoration using the `restorePreamble` to restore archived data:
+
+```typescript
+/**
+ * Simulates a restoration transaction to determine if restoration is needed.
+ * This function first checks the ledger entry for the given WASM hash. If the entry is found and has expired,
+ * it attempts a restoration. If the entry hasn't expired yet but the TTL needs extension, it proceeds with TTL extension.
+ * @param contract - The address of the contract to check
+ * @param txParams - The transaction parameters including account and signer.
+ * @returns A promise that resolves to a simulation response, indicating whether restoration or TTL extension is needed.
+ */
+export async function simulateRestorationIfNeeded(
+  contract: ContractAddress,
+  txParams: TxParams,
+): Promise<
+  SorobanRpc.Api.SimulateTransactionRestoreResponse | string | undefined
+> {
+  try {
+    const account = await config.rpc.getAccount(txParams.account.accountId());
+    const contract = new Contract(contract);
+    //const ledgerKey = getLedgerKeyWasmId(wasmHash);
```

Comment:
?

Reply:
> ?
removed that commented line, not sure why i left that in there.  Previously there was a helper function for this, but it can now be done with the getfootprint method of a Contract object.